### PR TITLE
Don’t panic on bad oauth and change server responses

### PR DESCRIPTION
**Pull request by theganyo**
XAPID-608, XAPID-612

### CODE REVIEW

File: apigee_sync.go [hunk 1]
```diff
@@ -213,67 +210,95 @@ func pollChangeAgent() error {
...
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		log.Error("Unable to read EdgeProxy Sever response ", err)
-		return false
-	}
+	for {
+		if first {
+			first = false
+		} else {
+			backOffFunc()
```
Comment (sramamoorthy73) [inline apigee_sync.go]: What happens if backOffFunc() max's?

File: apigee_sync.go [hunk 2]
```diff
@@ -213,67 +210,95 @@ func pollChangeAgent() error {
...
-	err = json.Unmarshal(body, &oauthResp)
-	if err != nil {
-		log.Error(err)
-		return false
-	}
-	token = oauthResp.AccessToken
+		token = ""
+		form := url.Values{}
+		form.Set("grant_type", "client_credentials")
+		form.Add("client_id", config.GetString(configConsumerKey))
```
Comment (sramamoorthy73) [inline apigee_sync.go]: Do we need to initialize 'req' each time we loop?

File: apigee_sync.go (hunk 1 above)
Comment (theganyo) [inline apigee_sync.go]: It continues to wait that max amount each time.

File: apigee_sync.go (hunk 2 above)
Comment (theganyo) [inline apigee_sync.go]: I'm not 100% positive it's necessary, but it's the safest thing to do so.

Comment (sramamoorthy73) [inline apigee_sync.go]: Hmmm. Ok.

Comment (theganyo) [inline apigee_sync.go]: Why is this a problem?

File: apigee_sync.go (hunk 1 above)
Comment (sramamoorthy73) [inline apigee_sync.go]: ok. There is no failure now.

File: apigee_sync.go (hunk 2 above)
Comment (sramamoorthy73) [inline apigee_sync.go]: No. I thought it was just redundant each time. May be looking only at the diff's, it unclear.

**Pull request merged**